labs/multiverse/src/widgets/room_view/details/read_receipts.rs

```diff
@@ -1,48 +1,81 @@
-    widgets::{Paragraph, Wrap},
+    widgets::{Block, Borders, HighlightSpacing, List, ListItem, ListState, Paragraph, Wrap},
 };
 
-use crate::TEXT_COLOR;
+use crate::{widgets::room_view::DetailsState, SELECTED_STYLE_FG, TEXT_COLOR};
 
 pub struct ReadReceipts<'a> {
-    room: Option<&'a Room>,
+    state: &'a DetailsState<'a>,
 }
 
 impl<'a> ReadReceipts<'a> {
-    pub fn new(room: Option<&'a Room>) -> Self {
-        Self { room }
+    pub(super) fn new(state: &'a DetailsState<'a>) -> Self {
+        Self { state }
     }
 }
 
+fn render_room_stats(room: &Room, area: Rect, buf: &mut Buffer) {
+    let RoomReadReceipts { num_unread, num_notifications, num_mentions, .. } = room.read_receipts();
+
+    let content = vec![
+        Line::from(format!("- unread: {num_unread}")),
+        Line::from(format!("- notifications: {num_notifications}")),
+        Line::from(format!("- mentions: {num_mentions}")),
+        Line::from(""),
+        Line::from("---"),
+        Line::from(format!("{:?}", room.read_receipts())),
+        Line::from("#"),
+    ];
+
+    Paragraph::new(content)
+        .fg(TEXT_COLOR)
+        .wrap(Wrap { trim: false })
+        .block(Block::new().borders(Borders::BOTTOM))
+        .render(area, buf);
+}
+
 impl Widget for &mut ReadReceipts<'_> {
     fn render(self, area: Rect, buf: &mut Buffer)
     where
         Self: Sized,
     {
-        match self.room {
-            Some(room) => {
-                let RoomReadReceipts { num_unread, num_notifications, num_mentions, .. } =
-                    room.read_receipts();
-
-                let content = vec![
-                    Line::from(format!("- unread: {num_unread}")),
-                    Line::from(format!("- notifications: {num_notifications}")),
-                    Line::from(format!("- mentions: {num_mentions}")),
-                    Line::from(""),
-                    Line::from("---"),
-                    Line::from(format!("{:?}", room.read_receipts())),
-                    Line::from("#"),
-                ];
-
-                Paragraph::new(content).fg(TEXT_COLOR).wrap(Wrap { trim: false }).render(area, buf);
-            }
-            None => {
-                let content = "(room disappeared in the room list service)";
-                Paragraph::new(content).fg(TEXT_COLOR).wrap(Wrap { trim: false }).render(area, buf);
+        if let Some(room) = self.state.selected_room {
+            match self.state.selected_item.as_deref() {
+                Some(selected_event) => {
+                    if let Some(item) = format_timeline_item(selected_event) {
+                        let list = List::new(vec![item])
+                            .highlight_spacing(HighlightSpacing::Always)
+                            .highlight_symbol(">")
+                            .highlight_style(SELECTED_STYLE_FG);
+
+                        let mut state = ListState::default();
+
+                        StatefulWidget::render(list, area, buf, &mut state);
```

Comment:
Any particular interesting in having this be a `List`, and not a plain `Paragraph`? As far as I can tell, there's no way to select the individual items in this list…

Reply:
No, it's just what it used to be when receipts for all items were shown.

Though I would imagine that we'd want to have prettier formatting for this, which might be nice to format using a list/table?